extensions/amp-story/1.0/amp-story-background-blur.js

```diff
@@ -0,0 +1,130 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {setImportantStyles} from '#core/dom/style';
+import {user} from '../../../src/log';
+
+/** @const {number} */
+const CANVAS_SIZE = 3;
+
+/** @const {number} */
+const DURATION_MS = 200;
+
+export class backgroundBlur {
+  /**
+   * @param {!Window} win
+   * @param {!Element} element
+   */
+  constructor(win, element) {
+    /** @private @const {!Window} */
+    this.win_ = win;
+
+    /** @private @const {!Element} */
+    this.element_ = element;
+
+    /** @private @const {!Element} */
+    this.canvas_ = null;
+
+    /**  @private {!number} */
+    this.currentRAF_ = null;
+  }
+
+  /**
+   * Setup canvas and attach it to the document.
+   */
+  attach() {
+    this.canvas_ = this.win_.document.createElement('canvas');
+    this.canvas_.width = this.canvas_.height = CANVAS_SIZE;
+    setImportantStyles(this.canvas_, {
+      width: '100%',
+      height: '100%',
+      position: 'absolute',
+      left: 0,
+      top: 0,
+    });
+    this.element_.appendChild(this.canvas_);
+  }
+
+  /**
+   * Remove canvas from the document and cancel the RAF.
+   */
+  detach() {
+    this.element_.removeChild(this.canvas_);
+    cancelAnimationFrame(this.currentRAF_);
+  }
+
+  /**
+   * Update the background to the specified page's background.
+   * @param {!Element} pageElement
+   */
+  update(pageElement) {
+    const fillElement = this.getBiggestImage_(pageElement);
+    if (!fillElement) {
+      user().info('No image found for background blur.');
+    }
+    this.animate_(fillElement);
+  }
+
+  /**
+   * Animated background transition.
+   * @private
+   * @param {?Element} fillElement
+   */
+  animate_(fillElement) {
+    const context = this.canvas_.getContext('2d');
+    let startTime;
+    const nextFrame = (currTime) => {
+      if (!startTime) {
+        startTime = currTime;
+      }
+      const elapsed = currTime - startTime;
+      if (elapsed < DURATION_MS) {
+        const easing = 1 - Math.pow(1 - elapsed / DURATION_MS, 2);
+        context.globalAlpha = easing;
+        context.fillRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);
+        if (fillElement) {
+          context.drawImage(fillElement, 0, 0, CANVAS_SIZE, CANVAS_SIZE);
+        }
+        this.currentRAF_ = requestAnimationFrame(nextFrame);
+      }
+    };
+    // Cancels the previous animation loop before starting a new one.
+    cancelAnimationFrame(this.currentRAF_);
+    this.currentRAF_ = requestAnimationFrame(nextFrame);
+  }
+
+  /**
+   * Get active page's largest amp-img element.
+   * @private
+   * @param {!Element} pageElement
+   * @return {?Element} An img element or null.
+   */
+  getBiggestImage_(pageElement) {
+    const getSize = (el) => {
+      if (el) {
+        const layoutBox = el.getLayoutBox();
+        return layoutBox.width * layoutBox.height;
+      } else {
+        return false;
+      }
```

Comment:
Just a personal preference and a complicated relationship with `else` statements, feel free to ignore 🥲

```suggestion
      if (!el) {
        return false;
      }
      const layoutBox = el.getLayoutBox();
      return layoutBox.width * layoutBox.height;
```

Reply:
This looks way cleaner to me. Thank you!